idflakies-core/src/main/java/edu/illinois/cs/dt/tools/detection/detectors/TuscanOnlyClassDetector.java

```diff
@@ -0,0 +1,61 @@
+package edu.illinois.cs.dt.tools.detection.detectors;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.illinois.cs.dt.tools.detection.DetectionRound;
+import edu.illinois.cs.dt.tools.detection.DetectorUtil;
+import edu.illinois.cs.dt.tools.detection.TestShuffler;
+import edu.illinois.cs.dt.tools.detection.filters.ConfirmationFilter;
+import edu.illinois.cs.dt.tools.detection.filters.UniqueFilter;
+import edu.illinois.cs.dt.tools.runner.InstrumentingSmartRunner;
+import edu.illinois.cs.testrunner.data.results.TestRunResult;
+import edu.illinois.cs.testrunner.runner.Runner;
+
+public class TuscanOnlyClassDetector extends ExecutingDetector {
+    private final List<String> tests;
+    private TestRunResult origResult;
+
+    private final TestShuffler testShuffler;
+
+    public static int getClassesSize(List<String> tests) {
+        List<String> classes = new ArrayList<String>();        
+        for (final String test : tests) {
+            final String className = TestShuffler.className(test);
+            if (!classes.contains(className)) {
+                classes.add(className);
+            }
+        }
+        return classes.size();
+    }
+    
+    public TuscanOnlyClassDetector(final Runner runner, final File baseDir, final int rounds, final String type, final List<String> tests) {
+        super(runner, baseDir, rounds, type);
+        int n = getClassesSize(tests);
+        if (n == 3 || n == 5) {
```

Comment:
Leave some comment here as a reminder that we need one more round than the number of test classes if n is 3 or 5.